source/includes/fundamentals/code-snippets/enterprise-auth.rs

```diff
@@ -0,0 +1,20 @@
+use mongodb::{ bson::doc, options::{ ClientOptions, Credential, AuthMechanism }, Client };
+
+#[tokio::main]
+async fn main() -> mongodb::error::Result<()> {
+    let uri = "<connection string>";
+    let mut client_options = ClientOptions::parse(uri).await?;
+
+    // start-ldap
+    let plain_cred = Credential::builder()
+        .username("<username>".to_string())
+        .password("<password>".to_string())
+        .mechanism(AuthMechanism::Plain)
```

Comment:
```suggestion
        .mechanism(AuthMechanism::Plain)
        .source("$external".to_string())
```

Suggest adding `$external` source. I expect it is needed to authenticate with LDAP. Use of `$external` is shown in the Java docs: https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/enterprise-auth/